gh-142927: Auto-open HTML output in browser after generation

ivonastojanovic opened the pull request:
Automatically open flamegraph and heatmap HTML output in the default browser after profiling completes. Add `--no-browser` flag to disable this behavior.

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNNN: Summary of the changes made
```

Where: gh-NNNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNNNN)
```

Where: [X.Y] is the branch name, for example: [3.13].

GH-NNNNNN refers to the PR number from `main`.

-->

Towards #142927.

CC: @pablogsal @lkollar

<!-- gh-issue-number: gh-138122 -->
* Issue: gh-138122
<!-- /gh-issue-number -->




<!-- gh-issue-number: gh-142927 -->
* Issue: gh-142927
<!-- /gh-issue-number -->

pablogsal commented on Lib/profiling/sampling/cli.py:
```diff
@@ -591,6 +597,32 @@ def _generate_output_filename(format_type, pid):
+    try:
+        webbrowser.open(file_url)
+    except Exception as e:
+        print(f"Warning: Could not open browser: {e}")
```
This should probably go to stderr

pablogsal commented on Lib/profiling/sampling/cli.py:
```diff
@@ -492,6 +493,11 @@ def _add_format_options(parser, include_compression=True, include_binary=True):
         "For heatmap: directory name (default: heatmap_PID)",
     )
+    output_group.add_argument(
+        "--no-browser",
```
We probably need to make this negative for all sub profilers if someone uses `--children` so only the main one does this

ivonastojanovic replied:
Ah good point

ivonastojanovic committed:
Add --browser flag to auto-open HTML output after generation

Add a --browser flag that automatically opens the generated
flamegraph and heatmap HTML files in the default browser once
profiling completes.

pablogsal reviewed (approved):
LGTM
